components/_Progress-Bar/progress-bar.js

```diff
@@ -0,0 +1,63 @@
+const msdsProgressBar = {
+  //Get the base number for the progress bar
+  parentElement: '',
+  baseNumber: 0,
+  currentStep: 0,
+  startFromOne: false,
+  init: function(parentElementId, baseNumber, customProgressionStep) {
+    this.parentElement = document.getElementById(parentElementId)
+
+    if (customProgressionStep > 0) {
+      this.currentStep = customProgressionStep
+    } else if (this.startFromOne == true) {
+      this.currentStep = 1
+    } else {
+      this.currentStep = 0
+    }
+    this.baseNumber = baseNumber
+    this.updateProgressbarDisplay()
+  },
+  updateProgressbarDisplay: function() { //update the display of the prorgess bar (step and counter)
```

Comment:
this should be broken up into 4 methods, something like:
function updateProgressBar(){
      calculatePercentage(this);
      updateSteps(this);
      updateCounter(this);
}

function calculatePercentage(progressBar){
   //calculate the percentage width of the progress step and update the look and fill base on this
   const updatedWidth = (this.currentStep * 100) / this.baseNumber
   progressBarProgressionElement.style.width = updatedWidth + '%'
}

function updateSteps(progressBar){
    if (this.currentStep == this.baseNumber) {
        progressBarProgressionElement.classList.add('msds-progress-bar__progression-bar--completed')
      } else {
        if (progressBarProgressionElement.classList.contains('msds-progress-bar__progression-bar--completed')) {
          progressBarProgressionElement.classList.remove('msds-progress-bar__progression-bar--completed')
        }
      }
}

function updateCounter(progressBar){
    const progressionCounterElement = this.parentElement.querySelector('.msds-progress-bar__progression-counter')
    progressionCounterElement.innerHTML = this.currentStep + '/' + this.baseNumber
}

just a rough explanation though, you will have to do tweaking